integration-tests/types/action.ts

```diff
@@ -0,0 +1,66 @@
+// import App, { ActionConstraints } from '../../src/App';
+// import { MessageAction } from '../../src/types';
+import { App } from '@slack/bolt';
+
+const app = new App({ token: 'TOKEN', signingSecret: 'Signing Secret' });
+
+// calling action method with incorrect an type constraint value should not work
+// $ExpectError
+app.action({ type: 'Something wrong' }, ({ action }) => {
+  return action;
+});
+
+/*
+// Should error because message_action doesn't have type action_id
+// $ Expect Error
+app.action({ type: 'message_action', action_id: 'dafasf' }, ({ action }) => {
+    // NOT WORKING
+    return action;
+});
+*/
+
+// Action in listner should be - MessageAction
+// $ExpectType void
+app.action({ type: 'message_action' }, ({
+  action, // $ExpectType MessageAction
+ }) => {
+  return action;
+});
+
+// $ExpectType void
+app.action({ type: 'block_actions' }, ({
+  action, // $ExpectType BlockElementAction
+ }) => {
+  return action;
+});
+
+// $ExpectType void
+app.action({ type: 'interactive_message' }, ({
+  action, // $ExpectType InteractiveAction
+ }) => {
+  return action;
+});
+
+// $ExpectType void
+app.action({ type: 'dialog_submission' }, ({
+  action, // $ExpectType DialogSubmitAction
+ }) => {
+  return action;
+});
+
+/* TODO: do we import MessageAction from local bolt?
```

Comment:
@aoberoi thoughts on what to do here? Does `bolt` export its `types`? Previously we did https://github.com/slackapi/bolt/pull/349/files#diff-4b3913c478f20245cef00a8912c57df3R2 to import `MessageAction`. Should I just use the local files in this repo to do so again? This does break the idea of having a fine division between this package and the local bolt copy to simulate the real world.

Reply:
Yup, I think you could add `MessageAction` to the import on L3:

```
import { App, MessageAction } from '@slack/bolt';
```

Reply:
derp. Thanks